lib/core/sdam/topology.js

```diff
@@ -925,20 +924,20 @@ function executeWriteOperation(args, options, callback) {
   const ns = args.ns;
   const ops = args.ops;
 
-  const willRetryWrite =
-    !args.retrying &&
-    !!options.retryWrites &&
-    options.session &&
-    isRetryableWritesSupported(topology) &&
-    !options.session.inTransaction() &&
-    options.explain === undefined;
-
   topology.selectServer(writableServerSelector(), options, (err, server) => {
     if (err) {
       callback(err, null);
       return;
     }
 
+    const willRetryWrite =
```

Comment:
On line 676 we have similiar logic to update and we discussed some code clarity while we're here:
```
    const currentlyNotRetrying = !args.retrying;
    const retryWrites = !!options.retryWrites;
    const hasSession = !!options.session;
    const supportsRetryableWrites = server.supportsRetryableWrites;
    const notInTransaction = !options.session.inTransaction();
    const notExplaining = options.explain === undefined;
    const willRetryWrite =
      currentlyNotRetrying &&
      retryWrites &&
      hasSession &&
      supportsRetryableWrites &&
      notInTransaction &&
      notExplaining;
```